tool/src/js/formatter.rs

```diff
@@ -43,6 +43,66 @@ const RESERVED: &[&str] = &[
     "with",
 ];
 
+/// From https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects
+const RESERVED_TYPES: &[&str] = &[
```

Comment:
I think this should also contain the lowercase identifiers like `undefined`, `eval`, etc., as they can be used as Diplomat type names as well.